sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonUtils.scala

```diff
@@ -55,4 +60,22 @@ object JacksonUtils {
 
     schema.foreach(field => verifyType(field.name, field.dataType))
   }
+
+  def strToStructType(schemaAsJson: String): StructType = Try {
+    DataType.fromJson(schemaAsJson).asInstanceOf[StructType]
+  }.getOrElse {
+    throw new AnalysisException(
+      s"""Illegal json string for representing a schema: $schemaAsJson"""")
```

Comment:
\"\"\"\" Four?

`s"Illegal json string for representing a schema: $schemaAsJson"`.